### PR TITLE
Make verbosity at DEBUG more reasonable No.2

### PR DESCRIPTION
**Pull request by fwolter**
A successful connection/disconnect in DEBUG looks like this now:
```
23:54:23.691 [DEBUG] [gnu.io.NativeResource                ] - Loaded native ARM library: libNRJavaSerialv8_HF
23:54:27.557 [DEBUG] [gnu.io.CommPortIdentifier            ] - Opened /dev/rs485-stick-upper (9600 8N1) by Modbus Serial Master
23:54:30.479 [DEBUG] [gnu.io.RXTXPort                      ] - Closing port: /dev/rs485-stick-upper
23:54:30.629 [DEBUG] [gnu.io.RXTXPort                      ] - Closed port: /dev/rs485-stick-upper
```

EDIT: During transmission, only TRACE messages are logged, except if errors occur.

### CODE REVIEW

**Comment by fwolter** [comment]
The GitHub Actions are pretty cool!

File: src/main/c/src/SerialImp.c
Comment (MrDOS) [inline src/main/c/src/SerialImp.c:4382]: I think the newline character is required here, right?

File: src/main/c/src/SerialImp.c
Comment (MrDOS) [inline src/main/c/src/SerialImp.c:5119]: I think changing the case of the method name here might've been a typo?

File: src/main/java/gnu/io/CommPortEnumerator.java
Comment (MrDOS) [inline src/main/java/gnu/io/CommPortEnumerator.java:81]: Ugh, why was this even here in the first place? Thank you for doing this.

File: src/main/java/gnu/io/CommPortIdentifier.java
Comment (MrDOS) [inline src/main/java/gnu/io/CommPortIdentifier.java:117]: Logging the exception directly will print its stack trace, right? Or are you intentionally suppressing that behaviour?

```suggestion
			log.error("Failed to load gnu.io.RXTXCommDriver:", e);
```

File: src/main/java/gnu/io/CommPortIdentifier.java
Comment (MrDOS) [inline src/main/java/gnu/io/CommPortIdentifier.java:165]: I think this can just come out altogether. I suspect there are really no trace-level logging statements we want to keep.

File: src/main/java/gnu/io/NativeResource.java
Comment (MrDOS) [inline src/main/java/gnu/io/NativeResource.java:86]: I agree that the newline is undesirable, but we probably want _some_ whitespace separation:

```suggestion
				throw new NativeResourceException("Unable to load native resource from given path: " + e.getLocalizedMessage());
```

File: src/main/java/gnu/io/RXTXCommDriver.java
Comment (MrDOS) [inline src/main/java/gnu/io/RXTXCommDriver.java:124]: `devel` is always false, so this is never printed. I'd just rip it out entirely. Nobody wants this to be printed automatically. If they want to log this information for their own diagnostic purposes, they can call `RXTXVersion.getVersion()`/`RXTXVersion.nativeGetVersion()` themselves.

File: src/main/java/gnu/io/RXTXCommDriver.java
Comment (MrDOS) [inline src/main/java/gnu/io/RXTXCommDriver.java:494]: Again, do you have a particular preference for suppressing the stack trace? I can see where there might be some concern about a library being too verbose with its errors.

```suggestion
					log.debug("Error reading the registry to get port list:", ex);
```

File: src/main/java/gnu/io/RXTXPort.java
Comment (MrDOS) [inline src/main/java/gnu/io/RXTXPort.java:960]: Think you probably meant to remove this trace, not reindent it.

File: src/main/java/gnu/io/RXTXPort.java
Comment (MrDOS) [inline src/main/java/gnu/io/RXTXPort.java:1193]: I think this is one of many good candidates for being replaced with a more sane, human-readable message. The presence of the method name in the output can be controlled by the log message formatter, and “monThreadisInterrupted” is fairly inscrutable to anyone not reading along with the code. That said, there are enough other messages which need rewriting that I think trying to do that in this PR (either yours or mine) will quickly spiral into a mess of highly subjective bikeshedding, so let's leave it for now.

File: src/main/java/gnu/io/Zystem.java
Comment (MrDOS) [inline src/main/java/gnu/io/Zystem.java:228]: This whole class exists to facilitate logging, but is not documented, and AFAIK, not used. That you've made changes here is fine, but the next step in refactoring logging is probably to remove this class altogether, so don't worry too much about any other changes to it.

**Review by MrDOS — APPROVED** [review]
This is really, really great. Thank you for doing this. I have a few comments, but they're changes I can easily apply myself. Let me know whether they're something you'd like to address or not, and then I'll either wait for your changes or merge right away.

Reading through these only makes me painfully aware of the remaining inconsistencies: some messages including method names, some not, some with punctuation, some starting with an upper-case letter, some lower-case... Not to mention that the whole library needs a formatting pass. But, one foot in front of the other. I'd like to get this into the library first, then focus on further cleanup.

Also, unless you object, after merging this into my feature branch, I'll probably squash it into your previous verbosity cleanup commit, then split the sum total into two commits: one for the Java code, and one for the native code, in keeping with how I'd organized the first few commits I made to the branch.

File: src/main/c/src/SerialImp.c
Comment (fwolter) [reply src/main/c/src/SerialImp.c:4382]: Why do you mean?

File: src/main/c/src/SerialImp.c
Comment (fwolter) [reply src/main/c/src/SerialImp.c:5119]: Well, I tried to make all starting characters upper case. IDK if that is reasonable here.

File: src/main/c/src/SerialImp.c
Comment (MrDOS) [reply src/main/c/src/SerialImp.c:4382]: Never mind, I got confused. I thought the format string needed a trailing newline for some reason. Forgot that it's the opposite, and that went to extra lengths to strip trailing newlines in the SLF4J wrapper.

File: src/main/java/gnu/io/CommPortIdentifier.java
Comment (fwolter) [reply src/main/java/gnu/io/CommPortIdentifier.java:117]: I log stack traces only if the method can be called from different places, i.e. if you really need the stack trace when investigating an error to identify the caller. But I'm fine if we log the stack trace here.

File: src/main/java/gnu/io/NativeResource.java
Comment (fwolter) [reply src/main/java/gnu/io/NativeResource.java:86]: Good catch!

File: src/main/c/src/SerialImp.c
Comment (MrDOS) [reply src/main/c/src/SerialImp.c:5119]: I like it in the other cases, but I think this one should stay lower-case because it's a reference to its own method name. Alternatively, you could use a real English sentence: e.g., “Event loop interrupted”.

File: src/main/java/gnu/io/RXTXCommDriver.java
Comment (fwolter) [reply src/main/java/gnu/io/RXTXCommDriver.java:124]: I think this comes from the old days when the JAR and the native libraries were shipped separately. I'd remove it, too.

File: src/main/java/gnu/io/RXTXPort.java
Comment (fwolter) [reply src/main/java/gnu/io/RXTXPort.java:960]: Yes, it's a tabs vs spaces issue.

**Comment by fwolter** [comment]
> Also, unless you object, after merging this into my feature branch, I'll probably squash it into your previous verbosity cleanup commit, then split the sum total into two commits: one for the Java code, and one for the native code, in keeping with how I'd organized the first few commits I made to the branch.

Yeah, sure!

I think it's easier for both of us if you apply the changes you think are reasonable. I'm willing to help to improve the code further, when we finished this raw cleanup.

File: src/main/java/gnu/io/CommPortIdentifier.java
Comment (MrDOS) [reply src/main/java/gnu/io/CommPortIdentifier.java:117]: Ah, right, that makes sense. Although I wonder how many people use a stack trace in a situation like this to quickly identify where _their_ code calls into the library. But let's keep it like this for now. I noticed another case of logging the message but not the stack trace, and it's probably the same situation as this.

File: src/main/c/src/SerialImp.c
Comment (fwolter) [reply src/main/c/src/SerialImp.c:5119]: I think a real sentence is a good idea :D

**Pull request merged**
